src/Donors/Models/Donor.php

```diff
@@ -0,0 +1,227 @@
+<?php
+
+namespace Give\Donors\Models;
+
+use DateTime;
+use Exception;
+use Give\Donations\ValueObjects\DonationMetaKeys;
+use Give\Donors\DataTransferObjects\DonorQueryData;
+use Give\Donors\Factories\DonorFactory;
+use Give\Framework\Exceptions\Primitives\InvalidArgumentException;
+use Give\Framework\Models\Contracts\ModelCrud;
+use Give\Framework\Models\Contracts\ModelHasFactory;
+use Give\Framework\Models\Factories\ModelFactory;
+use Give\Framework\Models\Model;
+use Give\Framework\QueryBuilder\QueryBuilder;
+
+/**
+ * Class Donor
+ *
+ * @unreleased
+ *
+ * @property int $id
+ * @property int $userId
+ * @property DateTime $createdAt
+ * @property string $name // TODO: name should be an accessor
+ * @property string $prefix
+ * @property string $firstName
+ * @property string $lastName
+ * @property string $email
+ */
+class Donor extends Model implements ModelCrud, ModelHasFactory
+{
+    /**
+     * @var string[]
+     */
+    protected $properties = [
+        'id' => 'int',
+        'userId' => 'int',
+        'createdAt' => DateTime::class,
+        'name' => 'string',
+        'firstName' => 'string',
+        'lastName' => 'string',
+        'email' => 'string',
+        'prefix' => 'string',
+    ];
+
+    /**
+     * @unreleased
+     *
+     * @param $id
+     *
+     * @return Donor
+     */
+    public static function find($id)
+    {
+        return give()->donorRepository->getById($id);
+    }
+
+    /**
+     * @unreleased
+     *
+     * @param  string  $email
+     * @return Donor
+     */
+    public static function whereEmail($email)
+    {
+        return give()->donorRepository->getByEmail($email);
+    }
+
+    /**
+     * @unreleased
+     *
+     * @param  string  $email
+     * @return bool
+     */
+    public function hasEmail($email)
+    {
+        $donor = give()->donorRepository->getByEmail($email);
+
+        return !is_null($donor);
+    }
+
+    /**
+     * @unreleased
+     *
+     * @param  int  $userId
+     * @return Donor
+     */
+    public static function whereUserId($userId)
+    {
+        return give()->donorRepository->getByWpUserId($userId);
+    }
+
+    /**
+     * @unreleased
+     *
+     * @param  string  $email
+     * @return bool
+     * @throws Exception
+     */
+    public function addAdditionalEmail($email)
+    {
+        return give()->donorRepository->insertAdditionalEmail($this->id, $email);
+    }
```

Comment:
This is breaking the model flow. This immediately persists, whereas models are supposed to enter hold persistence until `save()` is called.